File: src/Plexus/utils/Tasks/BorderTask.php
<?php

namespace Plexus\utils\Tasks;

use pocketmine\utils\TextFormat as C;

class BorderTask {
    
  /** @var string | Plexus\Main */
  private $plugin;
  
  /* 
   * Constructor
   */
  public function __construct(\Plexus\Main $plugin){
    $this->plugin = $plugin;
  }
  
  /*
   * Plugin
   * ===============================
   * - Returns $this->plugin = \Plexus\Main;
   * ===============================
   */
  public function getPlugin(){
    return $this->plugin;
  }
  
  /* 
   * Border
   * ===============================
   * - Task
   * ===============================
   */
  public function run(){
    $players = $this->getPlugin()->getServer()->getOnlinePlayers();
  foreach($players as $player){
    $spawn = $player->getLevel()->getSpawnLocation();
  if($player->getLevel()->getFolderName() === $this->getPlugin()->config()->spawn()){
  if(round($player->getPosition()->distance(new \pocketmine\math\Vector3($spawn->getX(), $spawn->getY() + 2, $spawn->getZ()))) >= $this->getPlugin()->config()->border or $player->getY() <= 2){
    $this->getPlugin()->spawn($player);
    $player->addTitle(C::YELLOW ."Border Reached!", C::RED ."You have reached the end of the world.", 50, 90, 40);
     }
    }
   } 
  }
}

File: src/Aether/tasks/borderTask.java
package Aether.tasks;

import Aether.AetherPlayer;
import Aether.Main;
import cn.nukkit.Player;
import cn.nukkit.utils.TextFormat;

public class borderTask extends handlerTask {

    private Main getPlugin() {
        return Main.getInstance();
    }

    @Override
    public void tick() {
        for (Player player : getPlugin().getServer().getOnlinePlayers().values()) {
            if (player != null) {
                if (player.getLevel() == getPlugin().getDefaultLevel()) {
                    if (player.getPosition().distance(getPlugin().getDefaultLevel().getSafeSpawn()) >= 120 || player.y <= 64) {
                        ((AetherPlayer) player).setupForTeleport();
                        player.teleport(getPlugin().getDefaultLevel().getSafeSpawn());

                        ((AetherPlayer) player).sendHub(false, TextFormat.RED + "Woah", TextFormat.RED + "You can't leave spawn silly :)");
                    }
                }
            }
        }
    }
}
